core/trino-main/src/main/java/io/trino/dispatcher/QueuedStatementResource.java

```diff
@@ -344,9 +345,9 @@ public long getLastToken()
             return lastToken.get();
         }
 
-        public boolean tryAbandonSubmissionWithTimeout(Duration querySubmissionTimeout)
+        public boolean tryAbandonSubmissionWithTimeout(long querySubmissionTimeoutNanos)
```

Comment:
Duration provides better type safety for this method. Is the instantiation of Duration in the body of this method a real problem?

Reply:
Did it cause actual GC problems (higher CPU usage by GC, GC pressure, etc)? Short-lived allocations are effectively "free", so the fact that they show up in the allocation profiling may be misleading.

Reply:
These could be promoted to old gen too (since we use these to track query progression)

Reply:
Yeah, this one does, but generally I've been looking at other places where we keep Durations for a longer period of time